### PR TITLE
Bump `torch` to 1.8.0

### PR DESCRIPTION
**Pull request by hvy**
## Motivation

Try bumping to latest torch releases.

## Description of the changes

See above.

### CODE REVIEW

**Comment by hvy** [comment]
```
#10 55.23 The conflict is caused by:
#10 55.23     optuna[checking,doctest,document,example,testing] 2.6.0.dev0 depends on torchvision==0.9.0+cpu
#10 55.23     allennlp 2.1.0 depends on torchvision<0.9.0 and >=0.8.1
#10 55.23     optuna[checking,doctest,document,example,testing] 2.6.0.dev0 depends on torchvision==0.9.0+cpu
#10 55.23     allennlp 2.0.1 depends on torchvision<0.9.0 and >=0.8.1
#10 55.23     optuna[checking,doctest,document,example,testing] 2.6.0.dev0 depends on torchvision==0.9.0+cpu
#10 55.23     allennlp 2.0.0 depends on torchvision<0.9.0 and >=0.8.1
```
https://github.com/optuna/optuna/pull/2442/checks?check_run_id=2035915426

**Comment by hvy** [comment]
Related ongoing PR https://github.com/optuna/optuna/pull/2442#issuecomment-791055048.

**Comment by hvy** [comment]
The tutorial that used to require the `urllib` hotfix now seems to work without it.

**Comment by himkt** [comment]
[Just a note]

I think AllenNLP will support torch v1.8.0 and torchvision v0.9.0 from the next release.
(ref. https://github.com/allenai/allennlp/commit/7f609901bcc58ba0050bb9a941c1fd7697614cdc)

**Comment by hvy** [comment]
Currently blocked by the following error.
```python
Traceback (most recent call last):
  File "examples/pytorch/pytorch_ignite_simple.py", line 133, in <module>
    study.optimize(objective, n_trials=100, timeout=600)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/optuna/study.py", line 394, in optimize
    show_progress_bar=show_progress_bar,
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/optuna/_optimize.py", line 76, in _optimize
    progress_bar=progress_bar,
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/optuna/_optimize.py", line 163, in _optimize_sequential
    trial = _run_trial(study, func, catch)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/optuna/_optimize.py", line 268, in _run_trial
    raise func_err
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/optuna/_optimize.py", line 217, in _run_trial
    value_or_values = func(trial)
  File "examples/pytorch/pytorch_ignite_simple.py", line 106, in objective
    train_loader, val_loader = get_data_loaders(TRAIN_BATCH_SIZE, VAL_BATCH_SIZE)
  File "examples/pytorch/pytorch_ignite_simple.py", line 75, in get_data_loaders
    train_data = MNIST(download=True, root=".", transform=data_transform, train=True)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/torchvision/datasets/mnist.py", line 79, in __init__
    self.download()
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/torchvision/datasets/mnist.py", line 146, in download
    download_and_extract_archive(url, download_root=self.raw_folder, filename=filename, md5=md5)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/torchvision/datasets/utils.py", line 314, in download_and_extract_archive
    download_url(url, download_root, filename, md5)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/torchvision/datasets/utils.py", line 140, in download_url
    raise e
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/torchvision/datasets/utils.py", line 132, in download_url
    _urlretrieve(url, fpath)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/torchvision/datasets/utils.py", line 29, in _urlretrieve
    with urllib.request.urlopen(urllib.request.Request(url, headers={"User-Agent": USER_AGENT})) as response:
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/urllib/request.py", line 641, in http_response
    'http', request, response, code, msg, hdrs)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 503: Service Unavailable
```

File: examples/chainer/chainer_integration.py [hunk 1]
```diff
@@ -23,13 +21,6 @@
 from optuna.trial import TrialState
 
 
-# TODO(crcrpar): Remove the below three lines once everything is ok.
-# Register a global custom opener to avoid HTTP Error 403: Forbidden when downloading MNIST.
-opener = urllib.request.build_opener()
-opener.addheaders = [("User-agent", "Mozilla/5.0")]
-urllib.request.install_opener(opener)
-
-
```
Comment (toshihikoyanase) [inline examples/chainer/chainer_integration.py:32]: I'm not sure if we can remove the UA of the HTTP requests to download MNIST in the example of non-pytorch examples including Chainer. Mayb we should discuss in https://github.com/optuna/optuna/issues/2469.

Comment (hvy) [reply examples/chainer/chainer_integration.py]: Thanks, you're absolutely right. I omitted the removals of these UA configurations.

**Comment by hvy** [comment]
PTAL.

**Comment by HideakiImamura** [comment]
@toshihikoyanase @himkt Could you review this PR?

**Review by HideakiImamura — APPROVED** [review]
LGTM. I think it makes sense to run the tests for non-allennlp first, and then run the tests for allennlp. These will be integrated after the next release of allennlp.

**Review by himkt — APPROVED** [review]
Let me merge this PR as we have a consensus that we should bump torch ASAP.

**Pull request merged**